### PR TITLE
It's a complete mess

### LINKED ISSUE #21: Fix PUT method function in game to update game history in the new API

### PR DESCRIPTION
**Pull request by mantisg**
This code has gotten out of hand. I can't wait to rewrite it, but I want to understand why what I'm doing here isn't working first. I need to know. If you can find why it's not working, let me know where the problem is and I'll look for myself and find the fix for it.

### CODE REVIEW

File: src/Components/App.jsx [hunk 1]
```diff
@@ -11,19 +11,17 @@ import '../styles.css'
...
 
     useEffect(() => {
         getAppData()
         .then(res => {
-            setAppData(res.content)
             setHistory(res.history)
-            setPlayers(res.players)
             setIsLoading(false)
         })
-    }, [])
```
Comment (marques-j-robinson) [inline src/Components/App.jsx:26]: This change is the issue.
[Documentation](https://codedamn.com/news/reactjs/useeffect-dependency)

**Review by marques-j-robinson — CHANGES_REQUESTED** [review]
I do see the error and have a fully functioning version on my computer right now.

**Comment by mantisg** [comment]
I can not believe I took out that dependency array and didn't put it back in. So silly

**Comment by mantisg** [comment]
Ok, so what I've done with this latest commit is try to separate history among all the games being played by setting the history when games are pulled to each game's specific history. However, the array that comes from the API is not being read. I have console logged it and isolated it. History logs as:
```JSON
[[null, null, null, null, null, null, null, null, null]]
```
When history is listed as:
```JSON
[null, null, null, null, null, null, null, null, null]
```
it works fine. When you add the wrapping array when there are no moves, the code can't process array functions on it like it should, considering the array of arrays format is how history ends up after some moves are made.

**Pull request merged**